File: backend/src/main/java/com/programmers/pcquotation/domain/estimaterequest/dto/EstimateRequestResDto.java
package com.programmers.pcquotation.domain.estimaterequest.dto;

import java.time.LocalDateTime;

import com.programmers.pcquotation.domain.estimaterequest.entity.EstimateRequestStatus;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Getter                 // JSON 직렬화를 위해 필요
@Builder
@NoArgsConstructor     // 기본 생성자
@AllArgsConstructor    // 모든 필드를 파라미터로 받는 생성자
public class EstimateRequestResDto {
	private Integer id;
	private String customerId;
	private EstimateRequestStatus status;
	private LocalDateTime createDate;
	private Integer budget;
	private String purpose;
	private String otherRequest;
}

File: backend/src/main/java/com/programmers/pcquotation/domain/estimaterequest/dto/EstimateRequestResDto.kt
package com.programmers.pcquotation.domain.estimaterequest.dto

import com.programmers.pcquotation.domain.estimaterequest.entity.EstimateRequest
import com.programmers.pcquotation.domain.estimaterequest.entity.EstimateRequestStatus
import java.time.LocalDateTime

data class EstimateRequestResDto(
    val id: Int,
    val customerName: String,
    val status: EstimateRequestStatus,
    val createDate: LocalDateTime,
    val budget: Int,
    val purpose: String,
    val otherRequest: String
) {
    constructor(estimateRequest: EstimateRequest) : this(
        id = estimateRequest.id,
        customerName = estimateRequest.customer.customerName,
        status = estimateRequest.status ?: EstimateRequestStatus.WAIT,
        createDate = estimateRequest.createDate,
        budget = estimateRequest.budget,
        purpose = estimateRequest.purpose,
        otherRequest = estimateRequest.otherRequest
    )
}